app/main.py

```diff
@@ -1,3 +1,84 @@
+import json
+import pathlib
+
+
 def shop_trip():
-    # write your code here
-    pass
+    data = parse_data()
+    fuel_price = data["FUEL_PRICE"]
+    customers = data["customers"]
+    shops = data["shops"]
+
+    for customer in customers:
+        print(f"{customer['name']} has {customer['money']} dollars")
+
+        shop, cost_of_way = choose_the_most_profitable_way(
+            customer,
+            shops,
+            fuel_price
+        )
+        if not shop:
+            continue
+        print("Date: 11/03/2020 13:15:34")
+        print(f"Thanks, {customer['name']}, for you purchase!")
+        bought_products(customer["product_cart"], shop["products"])
+
+        print(f"{customer['name']} rides home\n"
+              f"{customer['name']} now has "
+              f"{customer['money'] - cost_of_way:.2f} dollars\n")
+
+
+def parse_data():
+    BASE_DIR = pathlib.Path(__file__).resolve().parent
```

Comment:
Upper case is good for constants but there you have only local variable. And try to use more descriptive names for variables.